dspy/predict/react.py

```diff
@@ -95,12 +92,18 @@ def forward(self, **input_args):
         extract = self._call_with_potential_trajectory_truncation(self.extract, trajectory, **input_args)
         return dspy.Prediction(trajectory=trajectory, **extract)
 
-    async def aforward(self, **input_args):
+    async def aforward(self, additional_tools: Optional[list[Callable]] = None, **input_args):
         trajectory = {}
         max_iters = input_args.pop("max_iters", self.max_iters)
+        tools = self.tools | self._convert_tools(additional_tools)
```

Comment:
should we raise an error on duplicated tool instead of silent overriding since we call this `additional_tools`?

Reply:
Sounds good, added the validation